src/scripts/main.js

```diff
@@ -1,3 +1,263 @@
+
+  setTimeout(() => {
+    notification.remove();
+  }, 2000);
+};
+
+const validatePosition = (position) => {
+  if (
+    !position ||
+    !position.trim() ||
+    /\d/.test(position) ||
+    position.trim().length < 2
+  ) {
+    return false;
+  }
+
+  return true;
+};
+
+const formFields = {
+  name: {
+    label: 'Name:',
+    type: 'text',
+    qa: 'name',
+    required: true,
+  },
+  position: {
+    label: 'Position:',
+    type: 'text',
+    qa: 'position',
+    required: true,
+  },
+  office: {
+    label: 'Office:',
+    type: 'select',
+    qa: 'office',
+    required: true,
+    options: [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ],
+  },
+  age: {
+    label: 'Age:',
+    type: 'number',
+    qa: 'age',
+    required: true,
+  },
+  salary: {
+    label: 'Salary:',
+    type: 'number',
+    qa: 'salary',
+    required: true,
+  },
+};
+
+const form = document.createElement('form');
+
+form.className = 'new-employee-form';
+
+for (const field in formFields) {
+  const fieldConfig = formFields[field];
+  const label = document.createElement('label');
+
+  label.textContent = fieldConfig.label;
+
+  let input;
+
+  if (fieldConfig.type === 'select') {
+    input = document.createElement('select');
+
+    fieldConfig.options.forEach((optionValue) => {
+      const option = document.createElement('option');
+
+      option.value = optionValue;
+      option.textContent = optionValue;
+      input.appendChild(option);
+    });
+  } else {
+    input = document.createElement('input');
+    input.type = fieldConfig.type;
+  }
+
+  input.name = field;
+  input.setAttribute('data-qa', fieldConfig.qa);
+  label.appendChild(input);
+  form.appendChild(label);
+}
+
+const submitBtn = document.createElement('button');
+
+submitBtn.type = 'submit';
+submitBtn.textContent = 'Save to table';
+form.appendChild(submitBtn);
+document.body.appendChild(form);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = {};
+
+  for (const field in formFields) {
+    formData[field] = form.querySelector(`[name="${field}"]`).value.trim();
+  }
+
+  if (
+    !formData.name ||
+    !formData.position ||
+    !formData.office ||
+    !formData.age ||
+    !formData.salary
+  ) {
+    pushNotification('Error', 'All fields are required', 'error');
+
+    return;
+  }
+
+  if (formData.name.length < 4) {
+    pushNotification(
+      'Error',
+      'Name must be at least 4 characters long',
+      'error',
+    );
+
+    return;
+  }
+
+  if (!validatePosition(formData.position)) {
+    pushNotification(
+      'Error',
+      'Position must be at least 2 characters long and cannot contain numbers',
+      'error',
+    );
+
+    return;
+  }
+
+  const ageValue = Number(formData.age);
+
+  if (ageValue < 18 || ageValue > 90) {
+    pushNotification('Error', 'Age must be between 18 and 90', 'error');
+
+    return;
+  }
+
```

Comment:
Potential Issue: Ensure that `formData.salary` is a valid number before converting it to a number. Consider adding validation to check if the salary is a numeric value.